pkg/daemon/ceph/mds/filesystem.go

```diff
@@ -135,3 +140,87 @@ func DeleteFilesystem(context *clusterd.Context, clusterName, filesystemName str
 	logger.Infof("Removed file system %s", filesystemName)
 	return nil
 }
+
+// PrepareForDaemonUpgrade performs all actions necessary to ensure the filesystem is prepared
+// to have its daemon(s) updated. This helps ensure there is no aberrant behavior during upgrades.
+// If the mds is not prepared within the timeout window, an error will be reported.
+// Ceph docs: http://docs.ceph.com/docs/master/cephfs/upgrading/
+func (f *Filesystem) PrepareForDaemonUpgrade(
+	context *clusterd.Context,
+	clusterName string,
+	timeout time.Duration,
+) error {
+	// * Beginning of noted section 1
+	// This section is necessary for upgrading to Mimic and to/past Luminous 12.2.3.
```

Comment:
Okay. I think all of this code is still necessary since it is also appropriate for upgrading from Luminous to Mimic. My chat with Jan tomorrow morning will hopefully clear this up.